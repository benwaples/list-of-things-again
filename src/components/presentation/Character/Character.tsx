import React from 'react'
import { Link } from 'react-router-dom'

type Props = {
  id: string,
  name: string,
  imageUrl: string
}

export const Character = ({id, name, imageUrl}: Props) => {
  return (
    <div>
      <figure>
        <img src={imageUrl} alt={name} />
        <figcaption>{name}</figcaption>
      </figure>
      <Link to={`/characters/${id}`}>See More</Link>
    </div>
  )
}
