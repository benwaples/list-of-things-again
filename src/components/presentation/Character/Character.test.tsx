import React from 'react'
import { render } from '@testing-library/react'
import { Character } from './Character'
import { MemoryRouter } from 'react-router-dom'

describe('Character component', () => {
  
  it('should render a character', () => {
    const character = {
      id: 'string', 
      name: 'string', 
      imageUrl: 'string'
    }
    const { asFragment } = render(
      <MemoryRouter>
        <Character {...character} />
      </MemoryRouter>
    )

    expect(asFragment()).toMatchSnapshot()
  })
  
})
