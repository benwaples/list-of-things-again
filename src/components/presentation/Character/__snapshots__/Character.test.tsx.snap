// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character component should render a character 1`] = `
<DocumentFragment>
  <div>
    <figure>
      <img
        alt="string"
        src="string"
      />
      <figcaption>
        string
      </figcaption>
    </figure>
    <a
      href="/characters/string"
    >
      See More
    </a>
  </div>
</DocumentFragment>
`;
